## DMH - DATA

In [64]:
import pandas as pd
import json

**Story data**

Title: 	DMH - Pop-up_VR_Museum - Audio Recorded Stories 
- Description: 	The dataset contains transcribed audio-recorded stories one paragraph long of design objects in the Pop-up VR Museum. Translations are available at least in Finnish, English, and Swedish. 
- UUID: 	55b31e3a-0131-4f3b-ac80-04de70bfafde
- URL: https://spice.kmi.open.ac.uk/dataset/details/125
- Number of items: 122

In [65]:
audio_stories = pd.read_json('data/Audio Recorded Stories.json')
audio_stories

_id            Object name Object number  \
0    63738b5b619d8873905ef6fd  Coffee pot - Pehtoori          8182   
1    63738b5c78ff61337567e278  Coffee pot - Pehtoori          8182   
2    63738b5c78232564042adf4b  Coffee pot - Pehtoori          8182   
3    63738b7bfb9758706e727891                      *          B561   
4    63738b7a78ff61337567e28a      Vase - Aalto vase         41793   
..                        ...                    ...           ...   
117  63738b54461cf10d8073bc83       Canister – Jerry         32030   
118  63738b53fb9758706e727878       Canister - Jerry         32030   
119  63738b53f3d1b15c7d3cd1f0       Canister - Jerry         32030   
120  63738b53b6bab71ffc679aa4       Canister - Jerry         32030   
121  63738b52ec6863140217f367         Bicycle - Jopo         11780   

    Date of recording                  Context/event/workshop  \
0           15/3/2022  Workshop for a group, at Design Museum   
1           15/3/2022  Workshop for a group, at Design Museum   
2           15/3/2022  Workshop for a group, at Design Museum   
3          21/09/2022  Workshop for a group, at Design Museum   
4          15/03/2022  Workshop for a group, at Design Museum   
..                ...                                     ...   
117        27/09/2022                          Design Evening   
118        22/03/2022                               Laajasalo   
119        15/12/2021  Workshop for a group, at Design Museum   
120        15/12/2021  Workshop for a group, at Design Museum   
121        15/03/2022  Workshop for a group, at Design Museum   

              Contributor name  User ID         End-user community  \
0                    Anonymous   501851            Senior citizens   
1                    Anonymous   923304            Senior citizens   
2                    Anonymous   501851            Senior citizens   
3           Hamed Malek Rashed   317892  Finnish language students   
4                 Liisa Kemppi   608474            Senior citizens   
..                         ...      ...                        ...   
117                      Lauri   213442           General audience   
118             Maria Levander   789889            Senior citizens   
119  Päivi Castren-Kortekangas   789838            Senior citizens   
120  Päivi Castren-Kortekangas   789838            Senior citizens   
121          Eeva-Kaarina Öyry   751188            Senior citizens   

    Story ID(#) Original language   ... Additional info  \
0       #8182S5            Finnish  ...               *   
1       #8182S7            Finnish  ...               *   
2       #8182S6            Finnish  ...               *   
3       #B561S1            English  ...               *   
4      #41793S4            Finnish  ...               *   
..          ...                ...  ...             ...   
117    #32030S5            Finnish  ...               *   
118    #32030S1            Finnish  ...               *   
119    #32030S2            Finnish  ...               *   
120    #32030S3            Finnish  ...               *   
121   #751188S1            Finnish  ...               *   

                               _datasetid  _timestamp _timestamp_year  \
0    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668517300            2022   
1    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668517261            2022   
2    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668517231            2022   
3    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516731            2022   
4    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516730            2022   
..                                    ...         ...             ...   
117  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516692            2022   
118  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516691            2022   
119  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516691            2022   
120  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516691            2022   
121  55b31e3a-0131-4f3b-ac80-04de70bfafde  16685166

**Story data**

Title: 	DMH - Pop-up_VR_Museum - Written Stories 
- Description: 	The dataset contains written stories of 1-4 sentences of design objects in the Pop-up VR Museum. Translations are available at least in Finnish, English, and Swedish. 
- UUID: 	6eadb144-86e3-4617-998f-8cff93ca64a0
- URL: https://spice.kmi.open.ac.uk/dataset/details/124
- Number of items: 406

In [66]:
written_stories = pd.read_json('data/Written Stories.json')
written_stories

_id                    Object name Object number  \
0    636c14d40282857a3c150481          Windbreaker - Torstai         44166   
1    636c14d3b4028608e54d7427          Windbreaker - Torstai         44166   
2    636c14d39b33d34fc4602677          Windbreaker - Torstai         44166   
3    636c14d33f72b3416c26d33d          Windbreaker - Torstai         44166   
4    636c14d2e6c018559c4a401d          Windbreaker - Torstai         44166   
..                        ...                            ...           ...   
401  636c14499b33d34fc4602627  Armchair - Paimio armchair 41          9249   
402  636c1448566dcc2dea6212aa  Armchair - Paimio armchair 41          9249   
403  636c1448f5d07205d51f45c4  Armchair - Paimio armchair 41          9249   
404  636c1448b4028608e54d73d6  Armchair - Paimio armchair 41          9249   
405  636c14475d9061195e24abf3  Armchair - Paimio armchair 41          9249   

    Date of recording                  Context/event/workshop  \
0          22/03/2022                               Laajasalo   
1          15/03/2022  Workshop for a group, at Design Museum   
2          17/03/2022                                   Lahti   
3          22/03/2022                               Laajasalo   
4          19/01/2022                         Online workshop   
..                ...                                     ...   
401        30/08/2022                          Design Evening   
402        30/08/2022                          Design Evening   
403        30/08/2022                          Design Evening   
404        30/08/2022                          Design Evening   
405        30/08/2022                          Design Evening   

               Contributor name  User ID End-user community  Comment ID (#)  \
0                     Anonymous   838079     Senior citizens       #44166C7   
1             Eeva-Kaarina Öyry   751188     Senior citizens       #44166C4   
2    Riitta Katriina Louhensalo   788016     Senior citizens       #44166C5   
3                     Anonymous   601587     Senior citizens       #44166C6   
4                 Leena Wilkman   336160     Senior citizens       #44166C1   
..                          ...      ...                 ...            ...   
401                   Anonymous   906115    General audience        #9249C6   
402                   Anonymous   717077    General audience        #9249C2   
403                   Anonymous   623637    General audience        #9249C3   
404                   Anonymous   683378    General audience        #9249C4   
405                   Anonymous   171980    General audience        #9249C1   

    Original language   ... Other language translation Additional info  \
0              Finnish  ...                          *               *   
1              Finnish  ...                          *               *   
2              Finnish  ...                          *               *   
3              Finnish  ...                          *               *   
4              Finnish  ...                          *          online   
..                 ...  ...                        ...             ...   
401            Finnish  ...                          *               *   
402            English  ...                          *               *   
403            English  ...                          *               *   
404            English  ...                          *               *   
405            English  ...                          *               *   

                               _datasetid  _timestamp _timestamp_year  \
0    6eadb144-86e3-4617-998f-8cff93ca64a0  1668027604            2022   
1    6eadb144-86e3-4617-998f-8cff93ca64a0  1668027603            2022   
2    6eadb144-86e3-4617-998f-8cff93ca64a0  1668027603            2022   
3    6eadb144-86e3-4617-998f-8cff93ca64a0  1668027603            2022   
4    6eadb144-86e3-4617-998f-8cff93ca64a0  1668027602            2022   
..                              

In [67]:
# Check anonymous
result = audio_stories.loc[audio_stories['Contributor name'] != 'Anonymous']
result

_id        Object name Object number  \
3    63738b7bfb9758706e727891                  *          B561   
4    63738b7a78ff61337567e28a  Vase - Aalto vase         41793   
5    63738b7a4d6d36651d600e22  Vase - Aalto vase         41793   
6    63738b7aec6863140217f380  Vase - Aalto vase         41793   
7    63738b79461cf10d8073bc9a  Vase - Aalto vase         41793   
..                        ...                ...           ...   
117  63738b54461cf10d8073bc83   Canister – Jerry         32030   
118  63738b53fb9758706e727878   Canister - Jerry         32030   
119  63738b53f3d1b15c7d3cd1f0   Canister - Jerry         32030   
120  63738b53b6bab71ffc679aa4   Canister - Jerry         32030   
121  63738b52ec6863140217f367     Bicycle - Jopo         11780   

    Date of recording                  Context/event/workshop  \
3          21/09/2022  Workshop for a group, at Design Museum   
4          15/03/2022  Workshop for a group, at Design Museum   
5          17/03/2022                                   Lahti   
6          17/03/2022                                   Lahti   
7          08/03/2022                              Orimattila   
..                ...                                     ...   
117        27/09/2022                          Design Evening   
118        22/03/2022                               Laajasalo   
119        15/12/2021  Workshop for a group, at Design Museum   
120        15/12/2021  Workshop for a group, at Design Museum   
121        15/03/2022  Workshop for a group, at Design Museum   

               Contributor name  User ID         End-user community  \
3            Hamed Malek Rashed   317892  Finnish language students   
4                  Liisa Kemppi   608474            Senior citizens   
5    Riitta Katriina Louhensalo   788016            Senior citizens   
6                Terttu Valonen   677821            Senior citizens   
7                      Kirsti L   561211           General audience   
..                          ...      ...                        ...   
117                       Lauri   213442           General audience   
118              Maria Levander   789889            Senior citizens   
119   Päivi Castren-Kortekangas   789838            Senior citizens   
120   Päivi Castren-Kortekangas   789838            Senior citizens   
121           Eeva-Kaarina Öyry   751188            Senior citizens   

    Story ID(#) Original language   ... Additional info  \
3       #B561S1            English  ...               *   
4      #41793S4            Finnish  ...               *   
5      #41793S5            Finnish  ...               *   
6      #41793S6            Finnish  ...               *   
7      #41793S1            Finnish  ...               *   
..          ...                ...  ...             ...   
117    #32030S5            Finnish  ...               *   
118    #32030S1            Finnish  ...               *   
119    #32030S2            Finnish  ...               *   
120    #32030S3            Finnish  ...               *   
121   #751188S1            Finnish  ...               *   

                               _datasetid  _timestamp _timestamp_year  \
3    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516731            2022   
4    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516730            2022   
5    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516730            2022   
6    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516730            2022   
7    55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516729            2022   
..                                    ...         ...             ...   
117  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516692            2022   
118  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516691            2022   
119  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516691            2022   
120  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516691            2022   
121  55b31e3a-0131-4f3b-ac80-04de70bfafde  1668516690            2022   

    _timestam

In [68]:
userids = pd.unique(audio_stories['User ID'])
print(len(userids))
print(userids)

41
[501851 923304 317892 608474 788016 677821 561211 652225 631141 111333
 751188 221852 111222 133333 981129 766242 139120 523452 594472 766245
 436574 315375 234555 512341 964084 921848 254668 388721 451211 185280
 626889 434969 331897 789889 147199 924889 344521 845666 213442 119809
 789838]


In [69]:
userids = pd.unique(audio_stories['Contributor name'])
print(len(userids))
print(userids)

31
['Anonymous' 'Hamed Malek Rashed' 'Liisa Kemppi'
 'Riitta Katriina Louhensalo' 'Terttu Valonen' 'Kirsti L'
 'Marke Kaskimies' 'Sanjeewa Pawantha' 'Anneli Luoti' 'Eeva-Kaarina Öyry'
 'Ana Rodrigues' 'Pia Hintikka' 'Leena Riitinen' 'Kristiina Haara'
 'Kaarina Oinonen' 'Leena Lappalainen' 'Veikko Luoti' 'Helena' 'Sofi'
 'Fish' 'Tuuli Tiihonen' 'Pauliina Härkönen' 'Marja-Liisa'
 'Maria Levander' 'Anita Salmela' 'Elmeri Härkönen' 'Olli' 'Mirja. J'
 'Lauri' 'Saifuddin' 'Päivi Castren-Kortekangas']


In [70]:
userids = pd.unique(audio_stories['Context/event/workshop'])
print(len(userids))
print(userids)

7
['Workshop for a group, at Design Museum' 'Lahti' 'Orimattila'
 'Design Evening' 'Laajasalo' 'Maunula' 'Online workshop']


In [71]:
userids = pd.unique(audio_stories['End-user community'])
print(len(userids))
print(userids)

4
['Senior citizens' 'Finnish language students' 'General audience'
 'Asylum seekers']


In [72]:
# written stories
userids = pd.unique(written_stories['End-user community '])
print(len(userids))
print(userids)

4
['Senior citizens' 'General audience' 'Asylum seekers'
 'Finnish language students']


In [73]:
userids = pd.unique(written_stories['Context/event/workshop'])
print(len(userids))
print(userids)

8
['Laajasalo' 'Workshop for a group, at Design Museum' 'Lahti'
 'Online workshop' 'Orimattila' 'Maunula' 'Design Evening' nan]


In [74]:
userids = pd.unique(written_stories['Contributor name'])
print(len(userids))
print(userids)

28
['Anonymous' 'Eeva-Kaarina Öyry' 'Riitta Katriina Louhensalo'
 'Leena Wilkman' 'Fardoso' 'AnnaLiisa Joukainen' 'Marja Kalola' 'Khallat'
 'Leena Lappalainen' 'Liisa Kemppi' 'Olli' 'Kristiina Haara'
 'Leea Riittinen' 'Liisa Nummela' 'Soili Viitaniemi' 'Anita Salmela'
 'Anneli Hietanen' 'Päivi Castren-Kortekangas' 'Saleh'
 'Paivi Castren-Kortekangas' 'Seija Järvinen' 'Olli Jaakkola'
 'Lea Lifflander' 'A. R.' 'Halira' 'Leila Mustanoja-Syysmeri'
 'Päivi Castrén-Kortekangas' 'Marke Kaskimies']


In [75]:
import numpy as np

In [76]:
result = written_stories.loc[written_stories['Context/event/workshop'] == np.NaN]
result

Empty DataFrame
Columns: [_id, Object name, Object number, Date of recording, Context/event/workshop, Contributor name, User ID, End-user community , Comment ID (#), Original language , Finnish translation, English translation, Swedish translation, Other language translation, Additional info, _datasetid, _timestamp, _timestamp_year, _timestamp_month, _timestamp_day, _timestamp_hour, _timestamp_minute, _timestamp_second]
Index: []

[0 rows x 23 columns]

**Artwork data**

Title: 	DMH - Pop_up_VR_Museum - Objects classification - English
- Description: 	Classification of all objects from Design Museum Helsinki's permanent collection that are currently in the Pop-up VR Museum. The information about these objects is stored in this dataset in English.
- UUID: 	0daa0287-d7f4-4f03-a068-95f43afcc347
- URL: https://spice.kmi.open.ac.uk/dataset/details/111
- Number of items: 64

Each json entry includes the following information (they don't use ontologies):
- _id
- Object_name
- Object
- Special name
- Object number
- Designer
- Production_date
- Collection
- Manufacturer
- Dimension in cm: n x m x p format
- Weight in kg
- Material
- Colour
- Timestamp

In [77]:
import pandas as pd
import json

In [78]:
objects = pd.read_json('data/DMH - Pop_up_VR_Museum - Objects classification - English.json')
objects

_id                       Object name        Object  \
0   636bf322a8e46b2e35302848                 Ultima Thule 2332          bowl   
1   634f1907d8af1b50664089d0                                 *           jug   
2   634f1903a45b33465c10ceb2                       Windbreaker        Jacket   
3   634f1912d8af1b50664089d6           Rautaristi (Iron Cross)  wall textile   
4   634f1912e09bfd1d9a04b8f9  Valkoiset hevoset (White Horses)  wall textile   
..                       ...                               ...           ...   
59  634f18fee09bfd1d9a04b8ec              Paimio / Armchair 41         chair   
60  634f18fed4227a56290cb6c5                          Pastille         chair   
61  634f18feb7693f159a62e2ef                    High chair 616         chair   
62  634f18fd9d480d2bcc4ac79f                          Stool 60         chair   
63  634f18fdd8af1b50664089c9                            Pirkka         chair   

             Special name Object number                  Designer  \
0   fruit or dessert bowl         14219            Tapio Wirkkala   
1                  teapot          B561                         *   
2                  Jacket         44166         Kellokumpu, Ritva   
3               ryijy rug         16850  Keinänen-Baeckmann, Aino   
4               ryijy rug         13906           Sotavalta, Impi   
..                    ...           ...                       ...   
59               armchair          9249              Aalto, Alvar   
60               armchair         44165              Aarnio, Eero   
61             high chair          9226          Schultén, Ben af   
62                  stool         44163              Aalto, Alvar   
63            Small chair         44164        Tapiovaara, Ilmari   

   Production date      Collection  \
0             1970               *   
1     19th century          Antell   
2             1984  Käyttökokoelma   
3             1931               *   
4             1932               *   
..             ...             ...   
59            1932               *   
60            1968               *   
61            1965               *   
62            1933  Käyttökokoelma   
63            1955  Käyttökokoelma   

                                         Manufacturer  Dimension in cm  ...  \
0                   A Ahlström Oy, Iitalan lasitehdas          20x20x9  ...   
1                                                   *   16,8 x 14 x 15  ...   
2                                             Torstai                *  ...   
3                                       Oy Neovius Ab  145 cm x 100 cm  ...   
4                                 Humalajärvi, Kerttu  173 cm x 126 cm  ...   
..                                                ...              ...  ...   
59                    Artek, Huonekalutehdas Korhonen     60 x 64 x 80  ...   
60                                            Asko Oy     92 x 92 x 53  ...   
61  Artek Oy, Huonekalutehdas Korhonen Oy ja Norrc...   42,8 x 44 x 77  ...   
62                                              Artek     38 x 38 x 44  ...   
63                                        Artek-Vitra     76 x 41 x 38  ...   

                               Colour                            _datasetid  \
0                    colourless glass  0daa0287-d7f4-4f03-a068-95f43afcc347   
1             light brown, dark brown  0daa0287-d7f4-4f03-a068-95f43afcc347   
2           orange, green, blue, pink  0daa0287-d7f4-4f03-a068-95f43afcc347   
3   violet (centre), yellowish (edge)  0daa0287-d7f4-4f03-a068-95f43afcc347   
4                   white, red, brown  0daa0287-d7f4-4f03-a068-95f43afcc347   
..                                ...                                   ...   
59                       black, birch  0daa0287-d7f4-4f03-a068-95f43afcc347   
60                             yellow  0daa0287-d7f4-4f03-a068-95f43afcc347   
61                         red, birch  0daa0287-d7f4-4f03-a068-95f43afcc347   
62                              birch  

In [79]:
userids = pd.unique(objects['Object number'])
print(len(userids))
print(userids)

64
['14219' 'B561' '44166' '16850' '13906' '17876' '32628' '32627' '16851'
 '44184' '32361' '33288' '44181' '44182' '44183' '44178' '44179' '44180'
 '44175' '44176' '44177' '7762' 'ILM7606' '14218' '10026' '12595' '22769'
 'AM1699' '44185' 'A652' '44171, 44172, 44173' '44174' '8182' 'C370'
 '44168, 44169, 44170' '41816' '44167' '11780' '32271' '32030' '41379'
 '20260' '13154' '38267' '41581' '41377' '30074' '27303' '38518' '13081'
 '41821' '38467' '35971' '41768' 'B548' '17248' '9321' '41793' '16101'
 '9249' '44165' '9226' '44163' '44164']


In [80]:
objects_df = objects.copy()
objects_df['auxiliar'] = objects_df['Object number']

In [81]:
# Combine audio_stories with objects
stories_objects_df = pd.merge(audio_stories, objects_df, left_on='Object number', right_on='Object number', how='outer')
stories_objects_df

_id_x          Object name_x Object number  \
0    63738b5b619d8873905ef6fd  Coffee pot - Pehtoori          8182   
1    63738b5c78ff61337567e278  Coffee pot - Pehtoori          8182   
2    63738b5c78232564042adf4b  Coffee pot - Pehtoori          8182   
3    63738b5dec6863140217f36e  Coffee pot - Pehtoori          8182   
4    63738b5c4d6d36651d600e10  Coffee pot - Pehtoori          8182   
..                        ...                    ...           ...   
160                       NaN                    NaN         38467   
161                       NaN                    NaN          B548   
162                       NaN                    NaN         17248   
163                       NaN                    NaN         16101   
164                       NaN                    NaN          9249   

    Date of recording                  Context/event/workshop  \
0           15/3/2022  Workshop for a group, at Design Museum   
1           15/3/2022  Workshop for a group, at Design Museum   
2           15/3/2022  Workshop for a group, at Design Museum   
3           17/3/2022                                   Lahti   
4          22/03/2022                               Laajasalo   
..                ...                                     ...   
160               NaN                                     NaN   
161               NaN                                     NaN   
162               NaN                                     NaN   
163               NaN                                     NaN   
164               NaN                                     NaN   

    Contributor name   User ID End-user community Story ID(#)  \
0          Anonymous  501851.0    Senior citizens     #8182S5   
1          Anonymous  923304.0    Senior citizens     #8182S7   
2          Anonymous  501851.0    Senior citizens     #8182S6   
3     Terttu Valonen  677821.0    Senior citizens     #8182S9   
4     Maria Levander  789889.0    Senior citizens     #8182S8   
..               ...       ...                ...         ...   
160              NaN       NaN                NaN         NaN   
161              NaN       NaN                NaN         NaN   
162              NaN       NaN                NaN         NaN   
163              NaN       NaN                NaN         NaN   
164              NaN       NaN                NaN         NaN   

    Original language   ...                          _datasetid_y  \
0              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
1              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
2              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
3              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
4              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
..                 ...  ...                                   ...   
160                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
161                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
162                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
163                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
164                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   

     _timestamp_y _timestamp_year_y _timestamp_month_y _timestamp_day_y  \
0    1.666128e+09            2022.0               10.0             18.0   
1    1.666128e+09            2022.0               10.0             18.0   
2    1.666128e+09            2022.0               10.0             18.0   
3    1.666128e+09            2022.0               10.0             18.0   
4    1.666128e+09            2022.0               10.0             18.0   
..            ...               ...                ...              ...   
160  1.666128e+09            2022.0               10.0             18.0   
161  1.666128e+09            2022.0               10.0             18.0   
162  1.666128e+09            2022.0               10.0             18.0   
163  1.666128e+09            2022.0       

In [82]:
stories_objects_df.columns

Index(['_id_x', 'Object name_x', 'Object number', 'Date of recording',
       'Context/event/workshop', 'Contributor name', 'User ID',
       'End-user community', 'Story ID(#)', 'Original language ',
       'Audio recording file name', 'Start timestamp', 'End timestamp',
       'Finnish translation', 'English translation', 'Swedish translation',
       'Other language translation', 'Additional info', '_datasetid_x',
       '_timestamp_x', '_timestamp_year_x', '_timestamp_month_x',
       '_timestamp_day_x', '_timestamp_hour_x', '_timestamp_minute_x',
       '_timestamp_second_x', '_updated_x', '_id_y', 'Object name_y', 'Object',
       'Special name', 'Designer', 'Production date', 'Collection',
       'Manufacturer', 'Dimension in cm', 'Weight in kg', 'Material', 'Colour',
       '_datasetid_y', '_timestamp_y', '_timestamp_year_y',
       '_timestamp_month_y', '_timestamp_day_y', '_timestamp_hour_y',
       '_timestamp_minute_y', '_timestamp_second_y', '_updated_y', 'auxiliar'],
    

In [83]:
result = stories_objects_df.loc[stories_objects_df['Object number'] != stories_objects_df['auxiliar']]
result

_id_x      Object name_x Object number  \
80  63738b6378ff61337567e27c  Cutlery - Savonia         44168   
81  63738b634d6d36651d600e14  Cutlery - Savonia         44168   
82  63738b63ec6863140217f372  Cutlery - Savonia         44168   
83  63738b6278232564042adf4f  Cutlery - Savonia         44168   

   Date of recording                  Context/event/workshop  \
80        15/03/2022  Workshop for a group, at Design Museum   
81         15/3/2022  Workshop for a group, at Design Museum   
82         17/3/2022                                   Lahti   
83        22/03/2022                         Online workshop   

     Contributor name   User ID End-user community Story ID(#)  \
80  Leena Lappalainen  594472.0    Senior citizens    #44168S2   
81  Leena Lappalainen  594472.0    Senior citizens    #44168S3   
82     Terttu Valonen  677821.0    Senior citizens    #44168S4   
83          Anonymous  185280.0    Senior citizens    #44168S1   

   Original language   ... _datasetid_y _timestamp_y _timestamp_year_y  \
80            Finnish  ...          NaN          NaN               NaN   
81            Finnish  ...          NaN          NaN               NaN   
82            Finnish  ...          NaN          NaN               NaN   
83            Finnish  ...          NaN          NaN               NaN   

   _timestamp_month_y _timestamp_day_y _timestamp_hour_y _timestamp_minute_y  \
80                NaN              NaN               NaN                 NaN   
81                NaN              NaN               NaN                 NaN   
82                NaN              NaN               NaN                 NaN   
83                NaN              NaN               NaN                 NaN   

   _timestamp_second_y _updated_y  auxiliar  
80                 NaN        NaN       NaN  
81                 NaN        NaN       NaN  
82                 NaN        NaN       NaN  
83                 NaN        NaN       NaN  

[4 rows x 49 columns]

In [84]:
# Written stories
written_stories_df = written_stories.copy()
written_stories_df['auxiliar'] = written_stories_df['Object number']

In [85]:
# Combine audio_stories with objects
written_stories_objects_df = pd.merge(written_stories, objects_df, left_on='Object number', right_on='Object number', how='outer')
written_stories_objects_df

_id_x          Object name_x        Object number  \
0    636c14d40282857a3c150481  Windbreaker - Torstai                44166   
1    636c14d3b4028608e54d7427  Windbreaker - Torstai                44166   
2    636c14d39b33d34fc4602677  Windbreaker - Torstai                44166   
3    636c14d33f72b3416c26d33d  Windbreaker - Torstai                44166   
4    636c14d2e6c018559c4a401d  Windbreaker - Torstai                44166   
..                        ...                    ...                  ...   
409                       NaN                    NaN                44177   
410                       NaN                    NaN              ILM7606   
411                       NaN                    NaN                 A652   
412                       NaN                    NaN  44171, 44172, 44173   
413                       NaN                    NaN  44168, 44169, 44170   

    Date of recording                  Context/event/workshop  \
0          22/03/2022                               Laajasalo   
1          15/03/2022  Workshop for a group, at Design Museum   
2          17/03/2022                                   Lahti   
3          22/03/2022                               Laajasalo   
4          19/01/2022                         Online workshop   
..                ...                                     ...   
409               NaN                                     NaN   
410               NaN                                     NaN   
411               NaN                                     NaN   
412               NaN                                     NaN   
413               NaN                                     NaN   

               Contributor name   User ID End-user community  Comment ID (#)  \
0                     Anonymous  838079.0     Senior citizens       #44166C7   
1             Eeva-Kaarina Öyry  751188.0     Senior citizens       #44166C4   
2    Riitta Katriina Louhensalo  788016.0     Senior citizens       #44166C5   
3                     Anonymous  601587.0     Senior citizens       #44166C6   
4                 Leena Wilkman  336160.0     Senior citizens       #44166C1   
..                          ...       ...                 ...            ...   
409                         NaN       NaN                 NaN            NaN   
410                         NaN       NaN                 NaN            NaN   
411                         NaN       NaN                 NaN            NaN   
412                         NaN       NaN                 NaN            NaN   
413                         NaN       NaN                 NaN            NaN   

    Original language   ...                          _datasetid_y  \
0              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
1              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
2              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
3              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
4              Finnish  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
..                 ...  ...                                   ...   
409                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
410                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
411                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
412                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   
413                NaN  ...  0daa0287-d7f4-4f03-a068-95f43afcc347   

     _timestamp_y _timestamp_year_y _timestamp_month_y _timestamp_day_y  \
0    1.666129e+09            2022.0               10.0             18.0   
1    1.666129e+09            2022.0               10.0             18.0   
2    1.666129e+09            2022.0               10.0             18.0   
3    1.666129e+09            2022.0               10.0             18.0   
4    1.666129e+09            2022.0               10.0             18.0   
..            ...               ...                ...              ...   
409

In [88]:
result = written_stories_objects_df.loc[written_stories_objects_df['Object number'] != written_stories_objects_df['auxiliar']]
result

_id_x                    Object name_x Object number  \
104  636c14b20282857a3c15046d  Shot glass: i-103 (i-lasi 2003)         18903   
105  636c14b20e2530450149d6f9  Shot glass: i-103 (i-lasi 2003)         18903   
193  636c14930282857a3c15045b                Maternity package         40881   
194  636c14930e2530450149d6e7                Maternity package         40881   
292  636c146fb4028608e54d73ed                Cutlery - Savonia         44168   
293  636c146ee6c018559c4a3fe1                Cutlery - Savonia         44168   
294  636c146ef5d07205d51f45da                Cutlery - Savonia         44168   
295  636c146d0e2530450149d6d1                Cutlery - Savonia         44168   
296  636c146d22d37d085f130328                Cutlery - Savonia         44168   
297  636c146d5d9061195e24ac09                Cutlery - Savonia         44168   
298  636c146c3f72b3416c26d301                Cutlery - Savonia         44168   
299  636c146c0282857a3c150445                Cutlery - Savonia         44168   
300  636c146f6b245e0b9926dbcd                 Cutlery: Scandia         44171   

    Date of recording                  Context/event/workshop  \
104        25/10/2022                          Design Evening   
105        25/10/2022                          Design Evening   
193        25/10/2022                          Design Evening   
194        25/10/2022                          Design Evening   
292        22/03/2022                               Laajasalo   
293        17/03/2022                                   Lahti   
294        22/03/2022                               Laajasalo   
295        08/03/2022                              Orimattila   
296        15/03/2022  Workshop for a group, at Design Museum   
297        15/03/2022  Workshop for a group, at Design Museum   
298        19/01/2022                         Online workshop   
299        08/03/2022                              Orimattila   
300        25/10/2022                          Design Evening   

      Contributor name   User ID End-user community  Comment ID (#)  \
104          Anonymous  813076.0    General audience       #18903C1   
105          Anonymous  731226.0    General audience       #18903C2   
193          Anonymous  221852.0    General audience       #40881C1   
194          Anonymous  385914.0    General audience       #40881C2   
292          Anonymous  162617.0     Senior citizens      #44176C12   
293     Seija Järvinen  431604.0     Senior citizens      #44176C10   
294          Anonymous  185280.0     Senior citizens      #44176C11   
295          Anonymous  246939.0    General audience       #44176C7   
296          Anonymous  376777.0     Senior citizens       #44176C8   
297  Leena Lappalainen  594472.0     Senior citizens       #44176C9   
298      Olli Jaakkola  446753.0     Senior citizens       #44176C5   
299          Anonymous  535913.0    General audience       #44176C6   
300          Anonymous  627927.0    General audience       #44171C1   

    Original language   ... _datasetid_y _timestamp_y _timestamp_year_y  \
104            English  ...          NaN          NaN               NaN   
105            English  ...          NaN          NaN               NaN   
193            English  ...          NaN          NaN               NaN   
194            Finnish  ...          NaN          NaN               NaN   
292            Finnish  ...          NaN          NaN               NaN   
293            Finnish  ...          NaN          NaN               NaN   
294            Finnish  ...          NaN          NaN               NaN   
295            Finnish  ...          NaN          NaN               NaN   
296            Finnish  ...          NaN          NaN               NaN   
297            Finnish  ...          NaN          NaN               NaN   
298            Finnish  ...          NaN          NaN               NaN   
299            Finnish  ...          NaN          NaN               NaN   
300            Finnish

In [89]:
userids = pd.unique(result['Object number'])
print(len(userids))
print(userids)

4
['18903' '40881' '44168' '44171']


**DMH Stories**

- Title: 	DMH - Pop_up_VR_Museum - Written Stories
- Description: 	The dataset contains written stories of 1-4 sentences of design objects in the Pop-up VR Museum. Translations are available at least in Finnish, English, and Swedish.
- UUID: 	a6485ac3-fcdd-44f7-af05-58483999c944
- URL: https://spice.kmi.open.ac.uk/dataset/details/112
- Number of items: 367

Each json entry includes the following information:
- _id
- Object
- Object number
- Date of recording
- Context/event/workshop
- Contributor name
- End-user community
- Comment ID
- Original language
- Finnish, English, Swedish and other language translation
- Tags (keywords for the story)
- Additional_info
- Dataset_id
- Timestamp



In [45]:
stories_vr = pd.read_json('data/DMH - Pop_up_VR_Museum - Written Stories.json')
stories_vr

_id                          Object Object number  \
0    635066d3b4cb1601a61b6bbd  Garbage bin - City of Helsinki         44167   
1    6350672241842a53891d0068           Windbreaker - Torstai         44166   
2    635067223674d93942521554           Windbreaker - Torstai         44166   
3    63506722863e5377923506ad           Windbreaker - Torstai         44166   
4    635067211cd426747368f353           Windbreaker - Torstai         44166   
..                        ...                             ...           ...   
362  635066aa81c5fb497104d3c4   Armchair - Paimio armchair 41          9249   
363  635066aa0b12bb21103988c5   Armchair - Paimio armchair 41          9249   
364  635066a927e9e11b76553a27   Armchair - Paimio armchair 41          9249   
365  635066a941842a53891d001b   Armchair - Paimio armchair 41          9249   
366  635066a96661f85f2503eef7   Armchair - Paimio armchair 41          9249   

    Date of recording Context/event/workshop            Contributor name  \
0          17/03/2022                      *                   Anonymous   
1          17/03/2022                  Lahti  Riitta Katriina Louhensalo   
2          22/03/2022              Laajasalo                   Anonymous   
3          22/03/2022              Laajasalo                   Anonymous   
4          08/03/2022             Orimattila                   Anonymous   
..                ...                    ...                         ...   
362        30/08/2022         Design Evening                   Anonymous   
363        30/08/2022         Design Evening                   Anonymous   
364        30/08/2022         Design Evening                   Anonymous   
365        30/08/2022         Design Evening                   Anonymous   
366        30/08/2022         Design Evening                   Anonymous   

    End-user community  Comment ID (#) Original language   \
0       Senior citizens       #44167C2            Finnish   
1       Senior citizens       #44166C5            Finnish   
2       Senior citizens       #44166C6            Finnish   
3       Senior citizens       #44166C7            Finnish   
4      General audience       #44166C2            Finnish   
..                  ...            ...                ...   
362    General audience        #9249C5            Finnish   
363    General audience        #9249C6            Finnish   
364    General audience        #9249C1            English   
365    General audience        #9249C2            English   
366    General audience        #9249C3            English   

                                   Finnish translation  ... Additional info  \
0                                 Tämähän on ikuinen.   ...               *   
1    Tuulitakki on ikävimpiä vaatteita, joita suoma...  ...               *   
2                        Loputon pilan aihe vieläkin.   ...               *   
3    Tällainen tuulitakki/puku on aikoinaan ollut k...  ...               *   
4    Tuulipuvulla pääsi ilmaiseksi teatteriesitykse...  ...               *   
..                                                 ...  ...             ...   
362  Ensimmäinen vierailu Paimion parantolassa vaik...  ...               *   
363  Minun tarina. Kun olin lapsi, minä tykkään kot...  ...               *   
364                              Terapiaistuimia (?).   ...               *   
365  Hammaslääkärin tuoli. Painajaiseni 10-vuotiaan...  ...               *   
366                             Mukava? Vaikea sanoa.   ...               *   

                               _datasetid  _timestamp _timestamp_year  \
0    a6485ac3-fcdd-44f7-af05-58483999c944  1666214444            2022   
1    a6485ac3-fcdd-44f7-af05-58483999c944  1666213666            2022   
2    a6485ac3-fcdd-44f7-af05-58483999c944  1666213666            2022   
3    a6485ac3-fcdd-44f7-af05-58483999c944  1666213666            2022   
4    a6485ac3-fcdd-44f7-af05-58483999c944  1666213665            2022   
..                                    ...   

**Stories datasets with emotions**

Stories datasets for DMH: Narratives 304 stories (first 213 + second 91) stories from VR Museum (Helsinki) about design objects from workshop participants, classified by DEGARI (UNITO), Semantic Annotator (H-FARM), Putchik and SenticNET7 API.

Dataset 1 - 91 stories
- Name: DMH_91Stories_DEGARI_SemanticAnnotator
- UUID:  1396cecb-c11b-4f73-9b2c-7d4b01891c44
- Link:   https://spice.kmi.open.ac.uk/dataset/details/107

Dataset 2 - 213 stories
- Name:  DMH_213Stories_DEGARI_SemanticAnnotator
- UUID:   46ed6774-8af7-468c-b6a8-a0544d0b1ad2
- Link:     https://spice.kmi.open.ac.uk/dataset/details/104

In particular, each json-story is described by the following characteristics

    DEGARI emotions (emotions extracted by our system DEGARI-UNITO)
    Semantic Annotator
    Plutchik 8-emotions {anger, anticipation, disgust, fear, joy, sadness, surprise, trust} and its likelihood 
    concept parsing
    polarity classification
    intensity ranking
    emotion recognition
    aspect extraction
    personality prediction (OCEAN)
    sarcasm identification
    depression categorization
    toxicity spotting
    

In [46]:
stories_91 = pd.read_json('data/DMH_91Stories_DEGARI_SemanticAnnotator.json')
stories_91

_id  @id                                     dc:description  \
0     1    1  I have not appreciated any other set of cutler...   
1     2    2                               Seen at work (?).      
2     3    3  Grateful memories. Jopo fit into a small space...   
3     4    4  In this picture is a moped named Solifer that ...   
4     5    5  In the parking garage, when cleaning the car, ...   
..  ...  ...                                                ...   
86   87   87  And my grandmother had a complete set of Myrna...   
87   88   88  Stylish appearance. Colour options. Good pouri...   
88   89   89  The world of Moomins is thought-provoking – wh...   
89   90   90  Simplified, simple, somehow stylish, and light...   
90   91   91  Discussion: - From Solifer this comes to my mi...   

                                    dc:descriptionITA degari_emotions  \
0   Non ho apprezzato nessun altro set di posate d...              ,    
1                                Visto al lavoro (?).                   
2   Ricordi riconoscenti.Jopo si inserisce in un p...                   
3   In questa foto è un ciclomotore chiamato Solif...        Optimism   
4   Nel garage, quando si pulisce l'auto, non si p...                   
..                                                ...             ...   
86  E mia nonna aveva un set completo di miRNA nel...         Delight   
87  Aspetto elegante.Opzioni di colore.Buona esper...            Love   
88  Il mondo di Moomins è stimolante: quando ho av...              ,    
89  Semplificato, semplice, in qualche modo elegan...                   
90  Discussione:- Da solifer questo mi viene in me...            Love   

              caption                                  plutchik_emotions  \
0      SavoniaCutlery  {'anger': 0, 'anticipation': 0.243902439024390...   
1          Garbagebin  {'anger': 0, 'anticipation': 0, 'disgust': 0, ...   
2                Jopo  {'anger': 0, 'anticipation': 0.57471264367816,...   
3        SoliferMoped  {'anger': 0, 'anticipation': 0.131926121372031...   
4          Garbagebin  {'anger': 0, 'anticipation': 0, 'disgust': 0, ...   
..                ...                                                ...   
86           MyrnaCup  {'anger': 0, 'anticipation': 0.058479532163742...   
87  PehtooriCoffeePot  {'anger': 0, 'anticipation': 0.344827586206896...   
88          MoominMug  {'anger': 0, 'anticipation': 0.526315789473684...   
89               Jopo  {'anger': 0, 'anticipation': 0, 'disgust': 0, ...   
90       SoliferMoped  {'anger': 0, 'anticipation': 0, 'disgust': 0, ...   

        emotion_recognition  \
0         joy; pleasantness   
1            ecstasy; bliss   
2         bliss; enthusiasm   
3   enthusiasm; contentment   
4             grief; terror   
..                      ...   
86      delight; enthusiasm   
87          enthusiasm; joy   
88          grief; loathing   
89         loathing; terror   
90      ecstasy; enthusiasm   

                                      concept_parsing subjectivity_detection  \
0     [NOT 'appreciate'; 'buy'; 'wedding'; 'present']             SUBJECTIVE   
1                                            ['work']             SUBJECTIVE   
2   ['city'; 'bar'; 'grateful'; 'fit'; 'small'; 'c...             SUBJECTIVE   
3   ['car'; 'product'; 'feature'; 'quality'; 'devi...             SUBJECTIVE   
4                ['car'; 'parking garage'; 'garbage']             SUBJECTIVE   
..                                                ...                    ...   
86  ['table'; 'coffee'; 'put on table'; 'taken out...             SUBJECTIVE   
87  ['appearance'; 'coffee'; 'stylish'; 'colo'; 'g...             SUBJECTIVE   
88  ['thought provoking'; 'birthday'; 'employees';...             SUBJECTIVE   
89  [NOT 'long distance'; 'simple'; 'stylish'; 'li...             SUBJECTIVE   
90  ['school'; NOT 'know how'; 'discussion'; 'mind...             SUBJECTIVE   

    ...                            _datasetid  _timestamp _timestamp_year  

In [47]:
stories_213 = pd.read_json('data/DMH_213Stories_DEGARI_SemanticAnnotator.json')
stories_213

_id                Artefact Artefact_ID      Type  \
0      3  Garbage bin - Helsinki    Unlisted  response   
1      2        Fiskars scissors           3  response   
2      1           Solifer moped       32271  response   
3      8   Windbreaker - Torstai    Unlisted  response   
4      7               Rug - Suo    Unlisted  response   
..   ...                     ...         ...       ...   
208  208               Rug - Suo    Unlisted  response   
209  207          Jerry Canister       32029  response   
210  211        Fiskars scissors           3  response   
211  213  Garbage bin - Helsinki    Unlisted  response   
212  212           Solifer moped       32271  response   

                                               English  \
0    I always greet a a garbage bin with joy. The p...   
1    These were a part of my childhood home and now...   
2    In this picture there is a moped from the 1960...   
3    Yuck – the sweat smelled bad, when wearing a w...   
4    I am Kaarina and a very big friend of these lo...   
..                                                 ...   
208  And I have a picture of this rug from Ritva Pu...   
209  Now this one didn’t fit me at all. The comment...   
210  Orange is a primary colour. However, when you ...   
211                  These are nowadays full of masks.   
212  In this picture is a moped named Solifer that ...   

                                               Finnish  \
0    Roskista tervehdin aina ilolla. Saa tyhjättyä ...   
1    Kuulunut omaan lapsuuden kotiin kuin nyt omaan...   
2    Tässä on kuvassa tällanen mopedi 1960-luvulta....   
3                  Yäk – tuulipukuhiki haisi – nailon?   
4    Oon näitten kangaspuitten oikein suuri ystävä....   
..                                                 ...   
208  Ja mul on kuva tästä Ritva Puotilan ryijystä, ...   
209  Nyt ei loksahtanut ollenkaan. Nämä puheenvuoro...   
210  Oranssi on perusväri. Kuitenkin, kun näkee sen...   
211                   Nama on nykyisin taynna maskeja.   
212  Tässä kuvassa on Solifer-niminen mopedi, joka ...   

                                               Swedish  \
0    Jag hälsar alltid på en soptunna med glädje. F...   
1    Dessa var en del av mitt barndomshem och nu en...   
2    På den här bilden finns en moped från 1960-tal...   
3    Usch – svetten luktade illa, när man bar en vi...   
4    Jag är Kaarina och en mycket stor vän till väv...   
..                                                 ...   
208  Och jag har en bild på den här ryan från Ritva...   
209  Nu passade inte den här mig alls. Kommentarern...   
210  Orange är en primära färg. Men när du ser det,...   
211                   Dessa är numera fulla av masker.   
212  På den här bilden är en moped vid namn Solifer...   

                                               Italian  \
0    Saluto sempre un cestino con gioia.Le tasche p...   
1    Questi facevano parte della mia casa d'infanzi...   
2    In questa foto c'è un ciclomotore dagli anni '...   
3    Yuck - Il sudore aveva un cattivo odore, quand...   
4    Sono Kaarina e una grande amica di questi tela...   
..                                                 ...   
208  E ho una foto di questo tappeto di Ritva Puoti...   
209  Ora questo non mi andava affatto.I commenti qu...   
210  L'arancione è un colore primario.Tuttavia, qua...   
211                Questi sono oggi pieni di maschere.   
212  In questa foto è un ciclomotore chiamato Solif...   

            degari_emotions  \
0                             
1                             
2                             
3                             
4                       Joy   
..                      ...   
208  Disapproval; Pessimism   
209                           
210                           
211                           
212                           

                                     plutchik_emotions  ...  @id  \
0    {'anger': 0, 'anticipation': 0.454545454545454...  ...    3   
1    {'ange

**Stories datasets**

Title: 	DMH: Narratives 123 stories - DEGARI Classification
- Description: 	DMH: Narratives 123 stories about design objects from workshop participants, classified by DEGARI (UNITO)
- UUID: 	5b6b876d-ad95-4bd8-a27f-627a5a98fdb5
- URL: https://spice.kmi.open.ac.uk/dataset/details/95
- Number of items: 213


Title: 	DMH: Stories collected from users
- Description: 	Several stories of design objects collected from users during workshops.
- UUID: 	88abdd7d-73d9-4a75-902b-89c78f9df7c0
- URL: https://spice.kmi.open.ac.uk/dataset/details/91
- Number of items: 213
        
First dataset includesthe following information:
- id
- Artefact
- Artefact ID
- Type
- English, Finnish, Swedish, Italian
- degari_emotions

Second dataset includes the same information with degari_emotions replaced by the timestamp.

In [48]:
stories_123 = pd.read_json('data/DMH_Narratives 123 stories - DEGARI Classification.json')
stories_123

_id                Artefact Artefact_ID      Type  \
0    62d93a1b3f14fc4ec902c3f7                    Jopo       11780  response   
1    62d93a8d7a214e77ce077899                    Jopo       11780  response   
2    62da664ea9db90688000cd98  Garbage bin - Helsinki    Unlisted  response   
3    62d9cecede5b2b6ec450c38b         Jokapoika shirt       41821  response   
4    62da5bda4dd245740e1c97fe              Aalto Vase       41793  response   
..                        ...                     ...         ...       ...   
208  62d9b22a73ec1456ec3564e0           Solifer moped       32271  response   
209  62d935cedb959e6f244c942a                    Jopo       11780  response   
210  62d9c92ece4eac30a00db839        Fiskars scissors           3  response   
211  62d9cca3f093442d8f1b81ba         Jokapoika shirt       41821  response   
212  62da6072b899f448ad3e05e2   Windbreaker - Torstai    Unlisted  response   

                                               English  \
0    1. Unisex bicycle city culture  modern domesti...   
1    (Answering to the questions: What could be a c...   
2      A bad garbage bin, the trash spreads around it.   
3    A casual everyman’s shirt - not for celebrations.   
4         A classic still. 20 tulips fit well into it.   
..                                                 ...   
208  Well, there is this Solifer moped here, which ...   
209  When did the first Jopo bicycles come? This on...   
210  When I retired, I got a pair of green scissors...   
211  When I was young, I worked in the Marimekko st...   
212  Yuck – the sweat smelled bad, when wearing a w...   

                                               Finnish  \
0    Jopopyörä \n-”1. ‘unisex’ pyörä” \n- city-kult...   
1    (Vastauksena kysymyksiin: Mikä voisi olla esin...   
2                         Huono roskis, roskat leviää.   
3           Arkinen, jokapoikainen paita – ei juhlaan.   
4       Klassikko edelleen. 20 tulppaania istuu hyvin.   
..                                                 ...   
208  No täs on tämmöinen Solifer-mopo, joka tota no...   
209          Koska ne ekat Jopot tuli? Tämä on lasten.   
210  Kun jäin eläkkeelle, sain vihreät sakset. Kun ...   
211  Olen nuorena ollut kesätyössä Marimekon myymäl...   
212                Yäk – tuulipukuhiki haisi – nailon?   

                                               Swedish  \
0    - '1. \"Unisex\" cykel \n- stadskultur \n- mod...   
1    (Som svar på frågorna: Vad kunde vara en helt ...   
2     En dålig soptunna, soporna sprider sig runt den.   
3     En vardaglig allemans skjorta - inte för fester.   
4    En klassiker fortfarande. 20 tulpaner passar b...   
..                                                 ...   
208  Jo, det finns den här Solifer-mopeden här, som...   
209  När kom den första Jopo cyklarna? Den här är f...   
210  När jag gick i pension, fick jag grön sax. När...   
211  När jag var ung jobbade jag i Marimekko-butike...   
212  Usch – svetten luktade illa, när man bar en vi...   

                                               Italian     degari_emotions  
0    1. Bicicletta unisex città cultura domestica m...            Optimism  
1    (Rispondendo alle domande: quale potrebbe esse...           Curiosity  
2    Un cestino della spazzatura brutta, la spazzat...  Pessimism; Outrage  
3    Una camicia casual da uomo comune - non per le...             Delight  
4    Un classico ancora. 20 tulipani si adattano bene.                      
..                                                 ...                 ...  
208  Bene, c'è questa solitaria ciclistica qui, che...             Delight  
209  Quando sono arrivate le prime biciclette di Jo...                      
210  Quando mi sono ritirato, ho ricevuto un paio d...                      
211  Quando ero giovane, ho lavorato nel negozio Ma...                      
212  Yuck - Il sudore aveva un cattivo odore, quand...                      

[213 rows x 9 columns]

In [49]:
stories_users = pd.read_json('data/DMH_Stories collected from users.json')
stories_users

_id                artefact artefact_id      type  \
0    62da6754fe41a44fa40bf30e  Garbage bin - Helsinki    Unlisted  response   
1    62da67344dd245740e1c9837  Garbage bin - Helsinki    Unlisted  response   
2    62da6713a512c064a4577f14  Garbage bin - Helsinki    Unlisted  response   
3    62da66f4b899f448ad3e0603  Garbage bin - Helsinki    Unlisted  response   
4    62da66c9b968dd769b158e16  Garbage bin - Helsinki    Unlisted  response   
..                        ...                     ...         ...       ...   
208  62d939ac7a214e77ce077894                    Jopo       11780  response   
209  62d938bda8203560ba645320             Garbage bin   123456abc  response   
210  62d9382c7a214e77ce07788f                    Jopo       11780  response   
211  62d935cedb959e6f244c942a                    Jopo       11780  response   
212  62d933cee5725c41d852e699                    Jopo       11780  response   

                                               Finnish  \
0    Löysin Savonia-sarjaa kirpputorilta. Viisi eur...   
1                               Sirot, hyvät käteen.     
2    Ja minä aterimet esittelin täällä. Ja minulle ...   
3    Sitten tota mul on tällä hetkellä tässä. Koton...   
4    Ja tästä Savonia-sarjasta on vähän muistoja ja...   
..                                                 ...   
208  Jopopyörä \n- suomalainen \n- kaunis \n- näppä...   
209  Helkaman Jopo: \nKuvassa punaisena, mutta luul...   
210  Kiitollisia muistoja. Jopon sai pieneen tilaan...   
211         Koska ne ekat Jopot tuli? Tämä on lasten.    
212  Ja otetaas nyt muistoja tästä Joposta. Ja mull...   

                                               English  \
0    I found the Savonia series at the flea market....   
1                     Slender, fit well in the hand.     
2    And I presented the cutlery here. And for me, ...   
3    So at the moment, I have here. These Savonia c...   
4    And I have a few memories of this Savonia seri...   
..                                                 ...   
208  Jopo bicycle \n- Finnish \n- beautiful \n- han...   
209  Jopo of Helkama: \nIn the picture as red, but ...   
210  Grateful memories. Jopo fit into a small space...   
211  When did the first Jopo bicycles come? This on...   
212  And now let’s have some memories of this Jopo....   

                                               Swedish  \
0    Jag hittade Savonia-serien på loppisen. Fem eu...   
1                        Nätta, ligger bra i handen.     
2    Och jag presenterade besticken här. Och för mi...   
3    Så just nu har jag här. Dessa Savonia-bestick ...   
4    Och jag har några minnen av den här Savonia-se...   
..                                                 ...   
208  Jopo cykel \n- finska \n- skön \n- behändig at...   
209  Helkama’s Jopo: \nPå bilden som röd, men jag t...   
210  Tacksamma minnen. Jopo passar in i ett litet u...   
211  När kom den första Jopo cyklarna? Den här är f...   
212  Och nu ska vi ha några minnen av Jopo. Och Jop...   

                               _datasetid  _timestamp  _timestamp_year  \
0    88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658480468             2022   
1    88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658480436             2022   
2    88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658480403             2022   
3    88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658480372             2022   
4    88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658480329             2022   
..                                    ...         ...              ...   
208  88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658403244             2022   
209  88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658403005             2022   
210  88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658402860             2022   
211  88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658402254             2022   
212  88abdd7d-73d9-4a75-902b-89c78f9df7c0  1658401742             2022   

     _timestamp_month  _timestamp_day  _timestamp_hour  _timestamp_minute  \
0                   

## Connections between stories and objects (DMH - Pop_up_VR_Museum - Objects classification - English) 

DMH - Pop_up_VR_Museum - Written Stories
- **Object number**: links to **Object Number** in Objects classification.

DMH_91Stories_DEGARI_SemanticAnnotator.json
- Nothing linking to objects.
    
DMH_213Stories_DEGARI_SemanticAnnotator.json
- **Artefact**: links to **Object name** in Objects classification. 
- **Artefact_ID**: links to **Object number** in Objects classification. 

DMH_Stories collected from users / DMH_Narratives 123 stories - DEGARI Classification
- **artefact/Artefact**: links to **Object name** in Objects classification.
- **artefact_id/Artefact_ID**: links to **Object number** in Objects classification.



## Connections between users and interactions
DMH_Stories collected from users
- There is not information about the users/emotions

DMH_91Stories_DEGARI_SemanticAnnotator, DMH_213Stories_DEGARI_SemanticAnnotator, DMH_Narratives 123 stories - DEGARI Classification
- There is information about the emotions, but there is not any attribute(s) identifying the users.

DMH - Pop_up_VR_Museum - Written Stories
- The users are identified with the attribute **Contributor name**, but they cannot be linked to any emotion database.


 ## Summary
 
We lack information to complete the picture between:
- user
- artwork
- emotion or other interaction.

## Data Analysis - objects

In [24]:
print(len(pd.unique(objects['Designer'])))
print(pd.unique(objects['Designer']))

49
['*' 'Kellokumpu, Ritva' 'Keinänen-Baeckmann, Aino' 'Sotavalta, Impi'
 'Gallen-Kallela, Akseli' 'Simberg-Ehrström, Uhra' 'Brummer, Eva'
 'Raitio, A. W.' 'Leppänen, Helena' 'Puotila, Ritva'
 'Metsovaara, Marjatta' 'Procopé, Ulla  Tomula, Esteri'
 'Rintaniemi, Päivi' 'Alfström, Tony,  Keaney, Brian' 'Franck, Kaj'
 'Procopé, Ulla' 'Tarna, Tauno' 'Finch, Alfred William' 'Osol, Olga'
 'Holzer-Kjellberg, Friedl' 'Sarpaneva, Timo' 'Wirkkala, Tapio'
 'Hopea, Saara'
 'Jansson, Tove  Franck, Kaj  Slotte-Elevant, Tove (decoration)'
 'Bäckström, Olof,  Lindén, Olavi' 'Nurmesniemi, Antti' 'Babel, Adolf'
 'Rajalin, Börje,  Nurmesniemi, Antti' 'Creadesign Oy,   Kähönen, Hannu'
 'Hellman,   Rislakki, Eero (colours)' 'Lindh, Richard' 'Rislakki, Eero'
 'Pitkonen, Jorma' 'Saura, Samuli' 'Karjalainen, Heikki,  Jäntti, Mikko'
 'Weckström, Björn'
 'Ratia, Ristomatti (model of the bag),  Wakisaka, Katsuji (print)'
 'Johansson, Panu' 'Rimala, Annika' 'Nurmesniemi, Vuokko' 'Toikka, Oiva'
 'Koskinen, Harri' 

In [37]:
print(len(pd.unique(objects['Collection'])))
print(pd.unique(objects['Collection']))

6
['Antell' 'Käyttökokoelma' '*' 'Opetusministeriön ryijykokoelma'
 'Marimekko' 'Vuokko']


In [38]:
df = objects.copy()

df['Material'] = df['Material'].str.split(',')
df = df.explode('Material')

df['Colour'] = df['Colour'].str.split(',')
df = df.explode('Colour')

In [39]:
print(len(pd.unique(df['Material'])))
print(pd.unique(df['Material']))

33
['stoneware' 'polyesterivanu' 'wool' ' cotton' 'porcelain' 'ceramics'
 ' wood' 'plastic' 'glass' 'metal' ' metal' ' enamel' ' melamine'
 'cast iron' ' teak' 'fiberglass' ' leather' 'reinforced plastic'
 ' cast aluminium' 'silver' 'cotton' ' marquis' 'wood' 'printed cotton'
 'faience' 'pine' 'red clay' 'birch plywood' ' pine laminate (seat)'
 'birch' 'solid birch' ' plywood' 'fabric']


In [40]:
print(len(pd.unique(df['Colour'])))
print(pd.unique(df['Colour']))

45
['light brown' ' dark brown' 'orange' ' green' ' blue' ' pink'
 'violet (centre)' ' yellowish (edge)' 'white' ' red' ' brown' '?' 'black'
 ' white' 'brown' ' light pastel shade' 'blue'
 'different tones of red brown' 'red' ' aniline' ' yellow' 'yellow'
 ' wood' ' gold' 'green gray' 'blue / violet' 'colourless glass'
 'rubin red' ' purple' 'gray' ' black' 'clear' 'metal' 'green' 'silver'
 ' orange' ' violet' 'wood' 'blue tones' ' several colours' ' blue-green'
 'rio brown' ' birch' 'birch' 'pine']


## Data Analysis - users

In [51]:
stories_vr

_id                          Object Object number  \
0    635066d3b4cb1601a61b6bbd  Garbage bin - City of Helsinki         44167   
1    6350672241842a53891d0068           Windbreaker - Torstai         44166   
2    635067223674d93942521554           Windbreaker - Torstai         44166   
3    63506722863e5377923506ad           Windbreaker - Torstai         44166   
4    635067211cd426747368f353           Windbreaker - Torstai         44166   
..                        ...                             ...           ...   
362  635066aa81c5fb497104d3c4   Armchair - Paimio armchair 41          9249   
363  635066aa0b12bb21103988c5   Armchair - Paimio armchair 41          9249   
364  635066a927e9e11b76553a27   Armchair - Paimio armchair 41          9249   
365  635066a941842a53891d001b   Armchair - Paimio armchair 41          9249   
366  635066a96661f85f2503eef7   Armchair - Paimio armchair 41          9249   

    Date of recording Context/event/workshop            Contributor name  \
0          17/03/2022                      *                   Anonymous   
1          17/03/2022                  Lahti  Riitta Katriina Louhensalo   
2          22/03/2022              Laajasalo                   Anonymous   
3          22/03/2022              Laajasalo                   Anonymous   
4          08/03/2022             Orimattila                   Anonymous   
..                ...                    ...                         ...   
362        30/08/2022         Design Evening                   Anonymous   
363        30/08/2022         Design Evening                   Anonymous   
364        30/08/2022         Design Evening                   Anonymous   
365        30/08/2022         Design Evening                   Anonymous   
366        30/08/2022         Design Evening                   Anonymous   

    End-user community  Comment ID (#) Original language   \
0       Senior citizens       #44167C2            Finnish   
1       Senior citizens       #44166C5            Finnish   
2       Senior citizens       #44166C6            Finnish   
3       Senior citizens       #44166C7            Finnish   
4      General audience       #44166C2            Finnish   
..                  ...            ...                ...   
362    General audience        #9249C5            Finnish   
363    General audience        #9249C6            Finnish   
364    General audience        #9249C1            English   
365    General audience        #9249C2            English   
366    General audience        #9249C3            English   

                                   Finnish translation  ... Additional info  \
0                                 Tämähän on ikuinen.   ...               *   
1    Tuulitakki on ikävimpiä vaatteita, joita suoma...  ...               *   
2                        Loputon pilan aihe vieläkin.   ...               *   
3    Tällainen tuulitakki/puku on aikoinaan ollut k...  ...               *   
4    Tuulipuvulla pääsi ilmaiseksi teatteriesitykse...  ...               *   
..                                                 ...  ...             ...   
362  Ensimmäinen vierailu Paimion parantolassa vaik...  ...               *   
363  Minun tarina. Kun olin lapsi, minä tykkään kot...  ...               *   
364                              Terapiaistuimia (?).   ...               *   
365  Hammaslääkärin tuoli. Painajaiseni 10-vuotiaan...  ...               *   
366                             Mukava? Vaikea sanoa.   ...               *   

                               _datasetid  _timestamp _timestamp_year  \
0    a6485ac3-fcdd-44f7-af05-58483999c944  1666214444            2022   
1    a6485ac3-fcdd-44f7-af05-58483999c944  1666213666            2022   
2    a6485ac3-fcdd-44f7-af05-58483999c944  1666213666            2022   
3    a6485ac3-fcdd-44f7-af05-58483999c944  1666213666            2022   
4    a6485ac3-fcdd-44f7-af05-58483999c944  1666213665            2022   
..                                    ...   

In [52]:
print(len(stories_vr[stories_vr['Contributor name'] == 'Anonymous']))

326


In [61]:
print(len(pd.unique(stories_vr['End-user community '])))
print(pd.unique(stories_vr['End-user community ']))

4
['Senior citizens' 'General audience' 'Asylum seekers'
 'Finnish language students']


In [62]:
df = stories_vr[stories_vr['Contributor name'] != 'Anonymous']
df

_id                           Object Object number  \
1    6350672241842a53891d0068            Windbreaker - Torstai         44166   
6    6350672127e9e11b76553a74            Windbreaker - Torstai         44166   
9    63506720b4cb1601a61b6bef            Windbreaker - Torstai         44166   
20   6350671c863e5377923506a9      Vase - Kyllikki Salmenhaara         17248   
21   6350671c0b12bb211039890e      Vase - Kyllikki Salmenhaara         17248   
31   63506718e6c39c7dda113eda                Vase - Aalto vase         41793   
32   6350671827e9e11b76553a6e                Vase - Aalto vase         41793   
37   635067163674d9394252154c                Vase - Aalto vase         41793   
68   6350670c27e9e11b76553a66                     Subway bench         41816   
100  635067013674d9394252153e                Shirt - Jokapoika         41821   
101  63506701863e537792350697                Shirt - Jokapoika         41821   
108  635066ff1cd426747368f33d                Shirt - Jokapoika         41821   
119  635066fb863e537792350693               Scissors - Fiskars         44174   
122  635066fae6c39c7dda113ec6                        Rug - Suo         32361   
123  635066fa27e9e11b76553a5a               Scissors - Fiskars         44174   
124  635066f90b12bb21103988f6                        Rug - Suo         32361   
129  635066f8863e537792350691                        Rug - Suo         32361   
130  635066f71cd426747368f337             Pot - Timo Sarpaneva          C370   
138  635066f5863e53779235068f             Pot - Timo Sarpaneva          C370   
154  635066ef6661f85f2503ef22                 Mug - Moomin mug         44185   
157  635066eee6c39c7dda113ebe           Moped - Solifer export         32271   
159  635066ee41842a53891d0046                 Mug - Moomin mug         44185   
171  635066eae6c39c7dda113ebc           Moped - Solifer export         32271   
176  635066e86661f85f2503ef1e                     Lamp - Block         38467   
236  635066d441842a53891d0036   Garbage bin - City of Helsinki         44167   
238  635066d4863e53779235067b   Garbage bin - City of Helsinki         44167   
265  635066ca27e9e11b76553a3c                Cutlery - Savonia         44168   
270  635066c9e6c39c7dda113ea8                Cutlery - Savonia         44168   
272  635066c80b12bb21103988d8                Cutlery - Savonia         44168   
276  635066c7863e537792350673                      Cup - Myrna         44176   
290  635066c2b4cb1601a61b6bb1            Coffee pot - Pehtoori          8182   
302  635066be81c5fb497104d3d1            Cocktail glass – 2744         12595   
306  635066bd6661f85f2503ef04            Cocktail glass - 2744         12595   
310  635066bb81c5fb497104d3cf                 Chair - Pastille         44165   
311  635066bb0b12bb21103988d0                 Chair - Pastille         44165   
321  635066b83674d93942521537                 Chair - Pastille         44165   
324  635066b781c5fb497104d3cd                 Chair - Pastille         44165   
326  635066b627e9e11b76553a30  Ceramic tile - Viimeinen Ateria         35971   
334  635066b327e9e11b76553a2e                   Bicycle - Jopo         11780   
339  635066b2e6c39c7dda113e9a                   Bicycle - Jopo         11780   
342  635066b10b12bb21103988ca                   Bicycle - Jopo         11780   

    Date of recording                  Context/event/workshop  \
1          17/03/2022                                   Lahti   
6          15/03/2022  Workshop for a group, at Design Museum   
9          19/01/2022                         Online workshop   
20         21/09/2022  Workshop for a group, at Design Museum   
21         21/09/2022  Workshop for a group, at Design Museum   
31         17/03/2022                                     NaN   
32         17/03/2022                                     NaN   
37         19/01/2022                         Online workshop   
68         21/09/2022  Workshop for a group, at Design Museum   
100        15/0

In [63]:
print(len(pd.unique(df['Contributor name'])))
print(pd.unique(df['Contributor name']))

27
['Riitta Katriina Louhensalo' 'Eeva-Kaarina Öyry' 'Leena Wilkman'
 'Fardoso' 'AnnaLiisa Joukainen' 'Marja Kalola' 'Khallat'
 'Leena Lappalainen' 'Liisa Kemppi' 'Olli' 'Kristiina Haara'
 'Liisa Nummela' 'Leea Riittinen' 'Anita Salmela' 'Anneli Hietanen'
 'Soili Viitaniemi' 'Päivi Castren-Kortekangas' 'Saleh'
 'Paivi Castren-Kortekangas' 'Seija Järvinen' 'Olli Jaakkola'
 'Lea Lifflander' 'A. R.' 'Halira' 'Leila Mustanoja-Syysmeri'
 'Päivi Castrén-Kortekangas' 'Marke Kaskimies']


In [67]:
df['Contributor name'].value_counts()

Leena Wilkman                 4
Eeva-Kaarina Öyry             4
Leena Lappalainen             3
Olli                          3
Liisa Nummela                 2
Fardoso                       2
Anneli Hietanen               2
Halira                        2
Riitta Katriina Louhensalo    1
Seija Järvinen                1
Olli Jaakkola                 1
A. R.                         1
Lea Lifflander                1
Saleh                         1
Leila Mustanoja-Syysmeri      1
Päivi Castrén-Kortekangas     1
Paivi Castren-Kortekangas     1
Anita Salmela                 1
Päivi Castren-Kortekangas     1
Soili Viitaniemi              1
Leea Riittinen                1
Kristiina Haara               1
Liisa Kemppi                  1
Khallat                       1
Marja Kalola                  1
AnnaLiisa Joukainen           1
Marke Kaskimies               1
Name: Contributor name, dtype: int64

## Summary

Most contributions are made by anonymous users (326 out of 367) and we cannot differentiate between them.
All users (even the anonymous ones) provide the End-user community (['Senior citizens' 'General audience' 'Asylum seekers'
 'Finnish language students']).